core/src/test/java/com/datastax/oss/driver/internal/core/util/concurrent/ScheduledTimeoutCapturingTimer.java

```diff
@@ -0,0 +1,126 @@
+/*
+ * Copyright DataStax, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.datastax.oss.driver.internal.core.util.concurrent;
+
+import static org.assertj.core.api.Assertions.fail;
+
+import io.netty.util.Timeout;
+import io.netty.util.Timer;
+import io.netty.util.TimerTask;
+import java.util.Collections;
+import java.util.HashSet;
+import java.util.Set;
+import java.util.concurrent.ArrayBlockingQueue;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.atomic.AtomicBoolean;
+
+/**
+ * Implementation of Netty's {@link io.netty.util.Timer Timer} interface to capture scheduled {@link
+ * io.netty.util.Timeout Timeout} instead of running them, events so they can be run manually in
+ * tests.
+ */
+public class ScheduledTimeoutCapturingTimer implements Timer {
```

Comment:
Nit: I think we don't really need "Scheduled" in the name here (in the other class it refers to the fact that it captures tasks submitted via the `schedule` method). Maybe even `CapturingTimer` would be enough...

But it's more a matter of personal taste really, feel free to use what's most clear to you.